cove_iati/rulesets/iati_standard_v2_ruleset/steps/standard_ruleset_step_definitions.py

```diff
@@ -142,30 +148,48 @@ def step_not_expected(context, xpath_expression):
     return context, errors
 
 
-@then('`{attribute}` start date attribute must be chronologically before end date attribute')
+@then('`{xpath1}` must be chronologically before `{xpath2}`')
 @register_ruleset_errors()
-def step_start_date_before_end_date(context, attribute):
-    xpath1 = get_xobjects(context.xml, context.xpath_expression1)
-    xpath2 = get_xobjects(context.xml, context.xpath_expression2)
+def step_start_date_before_end_date(context, xpath1, xpath2):
+    parents = get_xobjects(context.xml, context.xpath_expression)
+
     errors = []
+    for parent in parents:
+        start_date_attrs = get_xobjects(parent, xpath1)
+        end_date_attrs = get_xobjects(parent, xpath2)
+
+        if not start_date_attrs or not end_date_attrs:
+            continue
+
+        start_date_attr = start_date_attrs[0]
+        end_date_attr = end_date_attrs[0]
 
-    if not xpath1 or not xpath2:
+        if not invalid_date_format(start_date_attr) and not invalid_date_format(end_date_attr):
+            start_date = datetime.datetime.strptime(start_date_attr, '%Y-%m-%d').date()
+            end_date = datetime.datetime.strptime(end_date_attr, '%Y-%m-%d').date()
+            if start_date >= end_date:
+                fail_msg = 'Start date ({}) must be before end date ({})'
+                path_attr1 = get_child_full_xpath(context.xml, start_date_attr)
+                path_attr2 = get_child_full_xpath(context.xml, end_date_attr)
+                errors.append({
+                    'message': fail_msg.format(start_date_attr, end_date_attr),
+                    'path': '{} & {}'.format(path_attr1, path_attr2)
+                })
+    return context, errors
+
+
+@then(u'`{attribute}` attribute must sum to 100')
```

Comment:
Is the scenario outline for this step missing (forgot to add a file)?

Reply:
d'oh

Reply:
I've rebased to include it.

Reply:
Cool. I just pulled the rebased branch and spotted a tiny typo in a feature file (older, NOT added by your code), I'm fixing and pushing it.